app/javascript/mastodon/features/compose/components/compose_form.js

```diff
@@ -153,7 +153,7 @@ class ComposeForm extends ImmutablePureComponent {
     //     - Replying to zero or one users, places the cursor at the end of the textbox.
     //     - Replying to more than one user, selects any usernames past the first;
     //       this provides a convenient shortcut to drop everyone else from the conversation.
-    if (this.props.focusDate && this.props.focusDate !== prevProps.focusDate) {
+    if ((this.props.focusDate && this.props.focusDate !== prevProps.focusDate) || this.props.autoFocus) {
```

Comment:
Added because the /share compose form also needs to move the cursor to the end

Reply:
Since this is called from `componentDidUpdate` as well, this will set the focus and selection to the very end of the textarea for every compose state change, including every keypress in either the textarea or the CW field. I would suggest setting the `focusDate` through a new action instead.

Reply:
I set ```focusDate``` in app/javascript/mastodon/reducers/compose.js instead of this change. How about it?
(Sorry, I am not familiar with the React...)